geode-connectors/src/acceptanceTest/java/org/apache/geode/connectors/jdbc/GfshJdbcMappingIntegrationTest.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.connectors.jdbc;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.geode.cache.CacheFactory;
+import org.apache.geode.cache.configuration.JndiBindingsType;
+import org.apache.geode.cache.configuration.RegionConfig;
+import org.apache.geode.cache.execute.Function;
+import org.apache.geode.cache.execute.ResultCollector;
+import org.apache.geode.connectors.jdbc.internal.cli.CreateMappingFunction;
+import org.apache.geode.connectors.jdbc.internal.cli.CreateMappingPreconditionCheckFunction;
+import org.apache.geode.connectors.jdbc.internal.configuration.FieldMapping;
+import org.apache.geode.connectors.jdbc.internal.configuration.RegionMapping;
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.management.configuration.RegionType;
+import org.apache.geode.management.internal.cli.commands.CreateJndiBindingCommand;
+import org.apache.geode.management.internal.cli.functions.CreateJndiBindingFunction;
+import org.apache.geode.management.internal.cli.functions.CreateRegionFunctionArgs;
+import org.apache.geode.management.internal.cli.functions.RegionCreateFunction;
+import org.apache.geode.management.internal.configuration.converters.RegionConverter;
+import org.apache.geode.management.internal.functions.CliFunctionResult;
+import org.apache.geode.management.internal.util.ManagementUtils;
+
+public class GfshJdbcMappingIntegrationTest extends JdbcMappingIntegrationTest {
+
+  @Override
+  protected InternalCache createCacheAndCreateJdbcMapping(String cacheXmlTestName)
+      throws Exception {
+    InternalCache cache =
+        (InternalCache) new CacheFactory().set("locators", "").set("mcast-port", "0").create();
+    Set<DistributedMember> targetMembers = findMembers(cache, null, null);
+
+    CliFunctionResult createRegionFuncResult = executeCreateRegionFunction(targetMembers);
```

Comment:
I understand that it is recommended to use GfshRule.
With this change, I would like to achieve the same behavior in gfsh command and cache.xml.
So I have implimented `JdbcMappingIntegrationTest` class to standardize tests.
I have used the internal functions because some parts of GfshRule can not verify in case of errors.

Reply:
This is an integration test, which tests the behavior of gfsh commands from a user's point view. I don't think this test should use internal functions of specific gfsh command's implementation. The users should not worry about the gfsh implementation.

And I don't recommend using a lot of `System.out.println` in the tests. Is that for debugging purpose?

If you have to use internal functions, I would recommend testing them in a unit test, or some other integration test or dunit that specifically test a specific gfsh command.  You said you use the internal functions because some parts of `GfshRule` can not verify in case of errors. Can you give a specific example?

Reply:
OK, I will try to implement the test using gfsh commands.

Reply:
I tried to implement the test using GfshRule, but I couldn't do it because I use `cache.getService(JdbcConnectorService.class)` to get the implementation of `JdbcConnectorService` to verify FieldMapping, and GfshRule does not allow it.
And I would like to make the condition setting part and the verification part of `CacheXmlJdbcMappingIntegrationTest` and` GfshXmlJdbcMappingIntegrationTest` common.

So I plan one of the following.
1. Change the `GfshJdbcMappingIntegrationTest` to `FunctionJdbcMappingIntegrationTest`.
2. Remove `GfshJdbcMappingIntegrationTest`. (I can use the `describe jdbc-mapping` command to verify it, but I don't think I should do that because it would reduce maintainability and make it difficult to commonize tests.)

Reply:
If you want to use `gfsh` and verify the field mapping, `CreateMappingCommandDUnitTest` has some good examples which use `GfshCommandRule` (not `GfshRule` though). You can see if you would like add some more tests there. Typically, a DUnit is used for testing that involves `gfsh` commands. In the DUnit, you create a cluster with locator and server(s) and execute the `gfsh` commands. 

Using the functions that implement a specific `gfsh` command in integration tests is discouraged. That is not what an integration test should do.

Reply:
I removed `GfshJdbcMappingIntegrationTest` because I no longer need to impliment it.